src/js/client/templateFormControl.js

```diff
@@ -0,0 +1,222 @@
+      a)  Any previous "success" messages are cleared.
+      b)  the error is transformed using the rules found in `options.rules.error`
+      c)  the results are passed along to the `error` subcomponent for display.
+  4.  If the AJAX request is successful:
+      a)  Any previous "error" messages are cleared.
+      b)  The results are transformed using the rules found in `options.rules.success` and passed to the `success` subcomponent for display.
+      c)  The results are transformed using the rules found in `options.rules.model`, and applied to the model using the change applier.
+
+ */
+/* global fluid, jQuery */
+(function ($) {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.templates.hb.client.templateFormControl");
+
+
+    gpii.templates.hb.client.templateFormControl.checkRequirements = function (that) {
+        var errors = [];
+
+        if (!that.options.templates || !that.options.templates.initial || !that.options.templates.error || !that.options.templates.success) {
+            errors.push("You have not configured any templates for use with this component.");
+        }
+
+        else if (!that.options.ajaxUrl) {
+            errors.push("Your AJAX options do not include a URL.  The AJAX request will not be able to complete as expected.");
+        }
+
+        if (!that.options.rules || !that.options.rules.error || !that.options.rules.success || !that.options.rules.submission) {
+            errors.push("You have not configured the required rules regarding how to handle AJAX responses.");
+        }
+
+        if (errors.length > 0) {
+            fluid.fail(errors);
+        }
+    };
+
+    gpii.templates.hb.client.templateFormControl.submitForm = function (that, event) {
+        if (event) {
+            event.preventDefault();
+        }
+
+        var options = fluid.copy(that.options.ajaxOptions);
+
+        var transformedModel = fluid.model.transformWithRules(that.model, that.options.rules.submission);
+
+        if (options.json) {
+            options.contentType = "application/json";
+        }
+
+        options.data = options.json ? JSON.stringify(transformedModel) : transformedModel;
+        $.ajax(options);
+    };
+
+    // Make sure that we end up with a reasonable JSON object before we apply our rules.
+    gpii.templates.hb.client.templateFormControl.jsonOrString = function (data) {
+        if (typeof data === "string") {
+            try {
+                var jsonData = JSON.parse(data);
+                return jsonData;
+            }
+            catch (e) {
+                // Treat the data as a string message if we can't otherwise parse it.
+                return { message: data };
+            }
+        }
+
+        return data;
+    };
+
+    gpii.templates.hb.client.templateFormControl.handleSuccess = function (that, data) {
+        var jsonData = gpii.templates.hb.client.templateFormControl.jsonOrString(data);
```

Comment:
This is tricksy and I predict it will lead to problems one day. It's better to be absolutely clear what the content type of your request is, and attempt to parse it as JSON exactly once if it is JSON.